browser/ipfs/ipfs_host_resolver_unittest.cc

```diff
@@ -239,8 +241,9 @@ TEST_F(IPFSHostResolverTest, ResolutionFailed) {
   ipfs_resolver.Resolve(
       net::HostPortPair(host, 11), net::NetworkIsolationKey(),
       net::DnsQueryType::TXT,
-      base::BindOnce([](const std::string& host, const std::string& dnslink) {
-        NOTREACHED();
+      base::BindOnce([](const std::string& host,
+                        const absl::optional<std::string>& dnslink) {
+        EXPECT_FALSE(dnslink);
```

Comment:
Same question, how do we ensure this is called?

Reply:
And for a few lines below, I see no points checking resolved_callback_called() if `HostResolvedCallback` is not bind at all.

Reply:
oh I see it's using SetCompleteCallbackForTesting to quit now, tho not sure we need that rather than just quit when this callback is run.

Reply:
Fixed